### ISSUE TITLE
Zammad not working after upgrading source installation to 5.0.x

### ISSUE THREAD

**Issue by alxndr42** [zammad/zammad#3901]
After upgrading my Zammad installation from 4.1.x to 5.0.3, the systemd service and processes are starting, but the application is not available (no sockets listening). The `production.log` shows no errors, but `scheduler_err.log` contains the following:

```
bundler: failed to load command: script/scheduler.rb (script/scheduler.rb)
/usr/share/rvm/gems/ruby-2.7.4/gems/zeitwerk-2.4.2/lib/zeitwerk/loader/callbacks.rb:18:in `on_file_autoloaded': expected file /opt/zammad/app/controllers/sessions/collection_base.rb to define constant Sessions::CollectionBase, but didn't (Zeitwerk::NameError)
	from /usr/share/rvm/gems/ruby-2.7.4/gems/zeitwerk-2.4.2/lib/zeitwerk/kernel.rb:27:in `block in require'
	from /usr/share/rvm/gems/ruby-2.7.4/gems/zeitwerk-2.4.2/lib/zeitwerk/kernel.rb:26:in `tap'
	from /usr/share/rvm/gems/ruby-2.7.4/gems/zeitwerk-2.4.2/lib/zeitwerk/kernel.rb:26:in `require'
	from /usr/share/rvm/gems/ruby-2.7.4/gems/zeitwerk-2.4.2/lib/zeitwerk/loader.rb:409:in `const_get'
	from /usr/share/rvm/gems/ruby-2.7.4/gems/zeitwerk-2.4.2/lib/zeitwerk/loader.rb:409:in `block (2 levels) in eager_load'
	from /usr/share/rvm/gems/ruby-2.7.4/gems/zeitwerk-2.4.2/lib/zeitwerk/loader.rb:733:in `block in ls'
	from /usr/share/rvm/gems/ruby-2.7.4/gems/zeitwerk-2.4.2/lib/zeitwerk/loader.rb:725:in `foreach'
	from /usr/share/rvm/gems/ruby-2.7.4/gems/zeitwerk-2.4.2/lib/zeitwerk/loader.rb:725:in `ls'
	from /usr/share/rvm/gems/ruby-2.7.4/gems/zeitwerk-2.4.2/lib/zeitwerk/loader.rb:404:in `block in eager_load'
	from /usr/share/rvm/gems/ruby-2.7.4/gems/zeitwerk-2.4.2/lib/zeitwerk/loader.rb:393:in `synchronize'
	from /usr/share/rvm/gems/ruby-2.7.4/gems/zeitwerk-2.4.2/lib/zeitwerk/loader.rb:393:in `eager_load'
	from /usr/share/rvm/gems/ruby-2.7.4/gems/zeitwerk-2.4.2/lib/zeitwerk/loader.rb:508:in `each'
	from /usr/share/rvm/gems/ruby-2.7.4/gems/zeitwerk-2.4.2/lib/zeitwerk/loader.rb:508:in `eager_load_all'
	from /usr/share/rvm/gems/ruby-2.7.4/gems/railties-6.0.4.1/lib/rails/application/finisher.rb:122:in `block in <module:Finisher>'
	from /usr/share/rvm/gems/ruby-2.7.4/gems/railties-6.0.4.1/lib/rails/initializable.rb:32:in `instance_exec'
	from /usr/share/rvm/gems/ruby-2.7.4/gems/railties-6.0.4.1/lib/rails/initializable.rb:32:in `run'
	from /usr/share/rvm/gems/ruby-2.7.4/gems/railties-6.0.4.1/lib/rails/initializable.rb:61:in `block in run_initializers'
	from /usr/share/rvm/rubies/ruby-2.7.4/lib/ruby/2.7.0/tsort.rb:228:in `block in tsort_each'
	from /usr/share/rvm/rubies/ruby-2.7.4/lib/ruby/2.7.0/tsort.rb:350:in `block (2 levels) in each_strongly_connected_component'
	from /usr/share/rvm/rubies/ruby-2.7.4/lib/ruby/2.7.0/tsort.rb:431:in `each_strongly_connected_component_from'
	from /usr/share/rvm/rubies/ruby-2.7.4/lib/ruby/2.7.0/tsort.rb:349:in `block in each_strongly_connected_component'
	from /usr/share/rvm/rubies/ruby-2.7.4/lib/ruby/2.7.0/tsort.rb:347:in `each'
	from /usr/share/rvm/rubies/ruby-2.7.4/lib/ruby/2.7.0/tsort.rb:347:in `call'
	from /usr/share/rvm/rubies/ruby-2.7.4/lib/ruby/2.7.0/tsort.rb:347:in `each_strongly_connected_component'
	from /usr/share/rvm/rubies/ruby-2.7.4/lib/ruby/2.7.0/tsort.rb:226:in `tsort_each'
	from /usr/share/rvm/rubies/ruby-2.7.4/lib/ruby/2.7.0/tsort.rb:205:in `tsort_each'
	from /usr/share/rvm/gems/ruby-2.7.4/gems/railties-6.0.4.1/lib/rails/initializable.rb:60:in `run_initializers'
	from /usr/share/rvm/gems/ruby-2.7.4/gems/railties-6.0.4.1/lib/rails/application.rb:363:in `initialize!'
	from /opt/zammad/config/environment.rb:7:in `<top (required)>'
	from script/scheduler.rb:59:in `require'
	from script/scheduler.rb:59:in `block in <top (required)>'
	from /usr/share/rvm/gems/ruby-2.7.4/gems/daemons-1.4.1/lib/daemons/application.rb:277:in `block in start_proc'
	from /usr/share/rvm/gems/ruby-2.7.4/gems/daemons-1.4.1/lib/daemons/daemonize.rb:84:in `call_as_daemon'
	from /usr/share/rvm/gems/ruby-2.7.4/gems/daemons-1.4.1/lib/daemons/application.rb:281:in `start_proc'
	from /usr/share/rvm/gems/ruby-2.7.4/gems/daemons-1.4.1/lib/daemons/application.rb:307:in `start'
	from /usr/share/rvm/gems/ruby-2.7.4/gems/daemons-1.4.1/lib/daemons/controller.rb:56:in `run'
	from /usr/share/rvm/gems/ruby-2.7.4/gems/daemons-1.4.1/lib/daemons.rb:199:in `block in run_proc'
	from /usr/share/rvm/gems/ruby-2.7.4/gems/daemons-1.4.1/lib/daemons/cmdline.rb:124:in `catch_exceptions'
	from /usr/share/rvm/gems/ruby-2.7.4/gems/daemons-1.4.1/lib/daemons.rb:198:in `run_proc'
	from script/scheduler.rb:55:in `<top (required)>'
	from /usr/share/rvm/gems/ruby-2.7.4/gems/bundler-2.3.0/lib/bundler/cli/exec.rb:58:in `load'
	from /usr/share/rvm/gems/ruby-2.7.4/gems/bundler-2.3.0/lib/bundler/cli/exec.rb:58:in `kernel_load'
	from /usr/share/rvm/gems/ruby-2.7.4/gems/bundler-2.3.0/lib/bundler/cli/exec.rb:23:in `run'
	from /usr/share/rvm/gems/ruby-2.7.4/gems/bundler-2.3.0/lib/bundler/cli.rb:484:in `exec'
	from /usr/share/rvm/gems/ruby-2.7.4/gems/bundler-2.3.0/lib/bundler/vendor/thor/lib/thor/command.rb:27:in `run'
	from /usr/share/rvm/gems/ruby-2.7.4/gems/bundler-2.3.0/lib/bundler/vendor/thor/lib/thor/invocation.rb:127:in `invoke_command'
	from /usr/share/rvm/gems/ruby-2.7.4/gems/bundler-2.3.0/lib/bundler/vendor/thor/lib/thor.rb:392:in `dispatch'
	from /usr/share/rvm/gems/ruby-2.7.4/gems/bundler-2.3.0/lib/bundler/cli.rb:31:in `dispatch'
	from /usr/share/rvm/gems/ruby-2.7.4/gems/bundler-2.3.0/lib/bundler/vendor/thor/lib/thor/base.rb:485:in `start'
	from /usr/share/rvm/gems/ruby-2.7.4/gems/bundler-2.3.0/lib/bundler/cli.rb:25:in `start'
	from /usr/share/rvm/gems/ruby-2.7.4/gems/bundler-2.3.0/exe/bundle:48:in `block in <top (required)>'
	from /usr/share/rvm/gems/ruby-2.7.4/gems/bundler-2.3.0/lib/bundler/friendly_errors.rb:103:in `with_friendly_errors'
	from /usr/share/rvm/gems/ruby-2.7.4/gems/bundler-2.3.0/exe/bundle:36:in `<top (required)>'
	from /usr/share/rvm/gems/ruby-2.7.4/bin/bundle:23:in `load'
	from /usr/share/rvm/gems/ruby-2.7.4/bin/bundle:23:in `<main>'
	from /usr/share/rvm/gems/ruby-2.7.4/bin/ruby_executable_hooks:22:in `eval'
	from /usr/share/rvm/gems/ruby-2.7.4/bin/ruby_executable_hooks:22:in `<main>'
```

All the steps in the `Updating Zammad` instructions before that seemed to work fine (no error messages).

### Infos:

* Used Zammad version: 5.0.3
* Installation method (source, package, ..): source
* Operating system: Ubuntu 20.04
* Database + version: PostgreSQL 10
* Elasticsearch version: none

### Expected behavior:

* Zammad should work after upgrading to 5.0.x

### Actual behavior:

* Zammad is not working after upgrading to 5.0.x

### Steps to reproduce the behavior:

* Install Zammad 4.1.x
* Follow instructions for upgrading to 5.0.x: https://docs.zammad.org/en/latest/install/update.html

Yes I'm sure this is a bug and no feature request or a general question.

**Comment by alxndr42** [comment]
FWIW, I tested a fresh installation of Zammad 5.0.3 in a local VM before that, and it seemed to be working fine. So this seems to be caused by the upgrade process somehow.

**Comment by alxndr42** [comment]
Interesting:

```bash
$ sha256sum zammad-*
bbae03268b6c26e1eb3b0bb307737089691f67307d6dedfdb8cc6376c9a06598  zammad-5.0.3-ftp.tar.gz
ce14c1f61fafff6504eb1c4fa8f11fe82b922fadfae204339cbec86b6897b292  zammad-5.0.3-github.tar.gz
1514063dcf77ce1b4c9a5e54ffce222a6108e6e8e694f6d279945ad481463923  zammad-stable.tar.gz
```

File sizes:

- zammad-5.0.3-ftp.tar.gz = 8169570 bytes
- zammad-5.0.3-github.tar.gz = 18912031 bytes
- zammad-stable.tar.gz = 18915410 bytes

**Comment by alxndr42** [comment]
I noticed that the file `/opt/zammad/app/controllers/sessions/collection_base.rb` from the error message exists in my upgraded installation, but not a fresh one. So I replaced the content of `/opt/zammad` with a fresh copy, restored `database.yml`, ran `rake assets:precompile` and started Zammad again, and this time the application was reachable.

I'll keep an eye on the logs, but it seems to me that this portion of the upgrade instructions might be wrong or misleading:

```bash
$ cd /opt
$ wget https://github.com/zammad/zammad/archive/stable.tar.gz
$ tar -xzf stable.tar.gz --strip-components 1 -C zammad
$ chown -R zammad:zammad zammad/
$ rm -f stable.tar.gz
```

My interpretation would be that you're supposed to untar the new relase on top of the existing installation?
